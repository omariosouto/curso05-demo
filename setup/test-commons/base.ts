import type {Config} from '@jest/types';

// Sync object
const config: Config.InitialOptions = {
  testEnvironment: 'jsdom',
  preset: 'ts-jest',
  verbose: true,
};
export default config;
