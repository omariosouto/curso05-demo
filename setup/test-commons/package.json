{
  "name": "@alura/test-commons",
  "version": "0.0.0",
  "private": true,
  "main": "index.js",
  "files": [
    "base.ts"
  ],
  "dependencies": {
    "jest": "27.0.0",
    "ts-jest": "^27.1.4",
    "ts-node": "^10.7.0"
  }
}
