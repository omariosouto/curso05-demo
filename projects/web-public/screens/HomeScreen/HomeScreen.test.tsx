import { screen, customRender } from '@alura/test-commons/react-testing-library';
import { HomeScreen } from "./HomeScreen";

const render = customRender();

describe("expect home", () => {
  it("expect home", () => {
    const { container } = render(<HomeScreen />);
    
    expect(container).toMatchSnapshot();
  });
});
