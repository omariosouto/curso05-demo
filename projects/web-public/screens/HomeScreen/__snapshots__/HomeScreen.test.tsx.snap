// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect home expect home 1`] = `
<div>
  <div>
    <h2>
      Home Screen do Projeto :)
    </h2>
    <p>
      Resultado da soma do commons: 
      4
    </p>
  </div>
</div>
`;
